src/main/java/handler/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package handler;
+
+import java.util.Map;
+import model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final Map<String, Integer> storage;
+
+    public PurchaseOperationHandler(Map<String, Integer> storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storage.put(transaction.getFruit(),
+                storage.getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Currently, you're directly updating the storage without any validation. You need to ensure that the quantity being purchased does not exceed the available quantity in the storage.